src/transactions/TransactionFrame.cpp

```diff
@@ -1622,6 +1622,14 @@ TransactionFrame::insertKeysForTxApply(UnorderedSet<LedgerKey>& keys) const
         }
         op->insertLedgerKeysToPrefetch(keys);
     }
+    if (isSoroban())
```

Comment:
Soroban prefetching should only be supported with BucketListDB, so need to change this to `if (isSoroban() && isUsingBucketsDB())`. We might need to through in a few more additional BucketsDB checks (and a few asserts might not hurt too).